src/globalInfo/accessors/aliasAccessor.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Nullable } from '@salesforce/ts-types';
+import { SfdxError } from '../../sfdxError';
+import { GlobalInfo } from '../globalInfoConfig';
+import { SfAliases, SfOrg, SfInfoKeys, SfToken } from '../types';
+
+export type Aliasable = string | Partial<SfOrg> | Partial<SfToken>;
+
+export class AliasAccessor {
+  public constructor(private globalInfo: GlobalInfo) {}
+
+  /**
+   * Returns all the aliases for all the usernames
+   */
+  public getAll(): SfAliases;
+  /**
+   * Returns all the aliases for a given entity
+   *
+   * @param entity the aliasable entity that you want to get the aliases of
+   */
+  public getAll(entity: Aliasable): string[];
+  public getAll(entity?: Aliasable): string[] | SfAliases {
+    const all = this.globalInfo.get(SfInfoKeys.ALIASES) || {};
+    if (entity) {
+      const username = this.getNameOf(entity);
+      return Object.entries(all)
+        .filter((entry) => entry[1] === username)
+        .map((entry) => entry[0]);
+    } else {
+      return all;
+    }
+  }
+
+  /**
+   * Returns the first alias found for a given entity
+   *
+   * @param entity the aliasable entity that you want to get the alias of
+   */
+  public get(entity: Aliasable): Nullable<string> {
+    return this.getAll(entity).find((alias) => alias) ?? null;
```

Comment:
Why use `.find()` here instead of just an array index?


```suggestion
    return this.getAll(entity)[0] ?? null;
```